cachito/workers/prune_archives.py

```diff
@@ -0,0 +1,217 @@
+    @limits(calls=api_calls_per_second, period=1)
+    def resolve_source_archive_ratelimited(archive: _ParsedArchive) -> Optional[_ResolvedArchive]:
+        return _resolve_source_archive(archive)
+
+    for parsed_archive in _get_parsed_source_archives(ARCHIVE_DIR):
+        resolved_archive = resolve_source_archive_ratelimited(parsed_archive)
+        if resolved_archive and resolved_archive.created < older_than:
+            yield resolved_archive
+
+
+def _get_latest_request(archive: _ParsedArchive) -> Optional[dict[str, Any]]:
+    """
+    Find the latest request matching the _ParsedArchive via the API.
+
+    Return None if no matching request is found.
+    """
+    url = f"{config.cachito_api_url.rstrip('/')}/requests/latest"
+    params = {
+        "repo_name": archive.repo_name,
+        "ref": archive.ref,
+    }
+
+    try:
+        response = session.get(url, params=params, timeout=config.cachito_api_timeout)
+        response.raise_for_status()
+    except requests.HTTPError:
+        if response.status_code == 404:
+            return None
+        log.error(
+            "The request to %s failed with the status code %d and the following text: %s",
+            url,
+            response.status_code,
+            response.text,
+        )
+        raise NetworkError("Failed to query the cachito API")
+    except requests.RequestException:
+        msg = f"The connection failed when querying {url}"
+        log.exception(msg)
+        raise NetworkError(msg)
+
+    return response.json()
+
+
+@app.callback()
+def configure_logging(verbose: bool = False):
+    """Configure logging for the app."""
+    log_level = logging.DEBUG if verbose else logging.INFO
+    handler = logging.StreamHandler()
+    handler.setFormatter(logging.Formatter(LOG_FORMAT))
+    log.setLevel(log_level)
+    log.addHandler(handler)
+
+
+def _validate_older_than(older_than: datetime) -> datetime:
+    """Ensure that the value of the --older-than CLI option is not more recent than the minimum."""
+    older_than_utc = older_than.replace(tzinfo=timezone.utc)
+    if older_than_utc > MINIMUM_AGE_DATETIME:
+        raise typer.BadParameter(f"cannot be more recent than {MINIMUM_AGE_DATETIME}")
+    return older_than_utc
+
+
+@app.command()
+def delete(
+    older_than: Annotated[
+        datetime,
+        typer.Option(
+            callback=_validate_older_than,
+            formats=["%Y-%m-%d"],
+            help="Deletes archives that are older than the specified date (UTC). YYYY-MM-DD",
+        ),
+    ] = DEFAULT_AGE_DATETIME,
+    api_calls_per_second: Annotated[
+        int, typer.Option(min=1, max=5, help="The API requests-per-second limit.")
+    ] = 1,
+    limit: Annotated[
+        Optional[int], typer.Option(min=1, help="The maximum number of stale archives to process.")
+    ] = None,
+    execute: Annotated[bool, typer.Option(help="Actual deletion will only occur if True.")] = False,
+):
+    """
+    List and delete stale source archives.
```

Comment:
Does it make more sense for `delete()` to call `list()`, rather than duplicating the code?

Reply:
Reworked this a bit in the final. Let me know if the changes work for you :+1: